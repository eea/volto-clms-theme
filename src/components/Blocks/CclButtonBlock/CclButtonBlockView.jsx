import React from 'react';
import CclButton from '@eeacms/volto-clms-theme/components/CclButton/CclButton';

const CclButtonBlockView = (props) => {
  const { data } = props;

  return (
    <CclButton
      url={
        data?.download && data?.href?.[0]?.['@type'] === 'File'
          ? data?.href?.[0]?.['@id'] + '/@@download/file'
          : data?.href?.[0]?.['@id']
      }
      disabled={data?.disabled}
      download={data?.download || data?.href?.[0]?.['@type'] === 'File'}
      target={
        data.target ||
        (data?.download && data.href[0]['@type'] === 'File' && '_blank')
      }
      mode={data.style}
    >
      {data.title || 'Text example...'}
    </CclButton>
  );
};

export default CclButtonBlockView;
