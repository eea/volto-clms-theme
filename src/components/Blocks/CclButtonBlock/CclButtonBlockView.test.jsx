import React from 'react';
import renderer from 'react-test-renderer';
import CclButtonBlockView from './CclButtonBlockView';

test('renders a CclButtonBlockView component', () => {
  const data = {
    href: [{ '@type': 'File', '@id': 'https://www.google.com' }],
    disabled: true,
    download: true,
  };
  const component = renderer.create(<CclButtonBlockView data={data} />);
  const json = component.toJSON();
  expect(json).toMatchSnapshot();

  const data2 = {
    href: [{ '@type': 'News Item', '@id': 'https://www.google.com' }],
    disabled: true,
  };
  const component2 = renderer.create(<CclButtonBlockView data={data2} />);
  const json2 = component2.toJSON();
  expect(json2).toMatchSnapshot();
});
