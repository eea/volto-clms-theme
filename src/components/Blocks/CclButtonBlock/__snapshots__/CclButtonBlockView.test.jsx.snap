// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a CclButtonBlockView component 1`] = `
<a
  className="ccl-button ccl-button--default"
  disabled={true}
  download={true}
  href="https://www.google.com/@@download/file"
  target="_blank"
>
  Text example...
</a>
`;

exports[`renders a CclButtonBlockView component 2`] = `
<a
  className="ccl-button ccl-button--default"
  disabled={true}
  download={false}
  href="https://www.google.com"
>
  Text example...
</a>
`;
